src/axom/sidre/core/Group.hpp

```diff
@@ -293,35 +295,38 @@ class Group
    */
   Group* setDefaultAllocator(umpire::Allocator alloc)
   {
-    m_default_allocator_id = alloc.getId();
+    setDefaultAllocator(alloc.getId());
     return this;
   }
+#endif
 
   /*!
    * \brief Set the default umpire::Allocator associated with this Group.
    */
   Group* setDefaultAllocator(int allocId)
   {
+#if !defined(AXOM_USE_UMPIRE)
+    SLIC_ASSERT(allocId == axom::MALLOC_ALLOCATOR_ID);
```

Comment:
Is this assert necessary?
It's checking the value of `allocId`, but not doing anything with it.